test/e2e/aws.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	// AwsCredsSecretIDKey is secret key containing AWS KeyId
+	AwsCredsSecretIDKey = "awsAccessKeyId"
+	// AwsCredsSecretAccessKey is secret key containing AWS Secret Key
+	AwsCredsSecretAccessKey = "awsSecretAccessKey"
+)
+
+// NewClient creates our client wrapper object for the actual AWS clients we use.
+// For authentication the underlying clients will use either the cluster AWS credentials
+// secret if defined (i.e. in the root cluster),
+// otherwise the IAM profile of the master where the actuator will run. (target clusters)
+func NewClient(kubeClient kubernetes.Interface, secretName, namespace, region string) (*AWSClient, error) {
```

Comment:
Let's reuse the client we have under https://github.com/openshift/cluster-api-provider-aws/blob/master/cloud/aws/client/client.go